cpp/src/gandiva/function_registry_string.cc

```diff
@@ -216,6 +216,10 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
                      "concat_utf8_utf8_utf8_utf8_utf8_utf8_utf8_utf8_utf8_utf8",
                      NativeFunction::kNeedsContext),
 
+      NativeFunction("byte_substr", {"bytesubstring"},
+                     DataTypeVector{binary(), int32(), int32()}, binary(),
+                     kResultNullIfNull, "byte_substr", NativeFunction::kNeedsContext),
```

Comment:
nit: other function signature keep the argument types in the function name. Lets keep that convention